host/xtest/regression_1000.c

```diff
@@ -1879,3 +1879,74 @@ static void xtest_tee_test_1023(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1023, xtest_tee_test_1023,
 		"Test ELF initialization (.init_array)");
+
+static void xtest_tee_test_1024(ADBG_Case_t *c)
+{
+	TEEC_Session session = { 0 };
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig = 0;
+	uint8_t *empty_buf = NULL;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		xtest_teec_open_session(&session, &os_test_ta_uuid, NULL,
+			&ret_orig)))
+		return;
+
+	empty_buf = malloc(0);
+	if (!empty_buf) {
+		(void)ADBG_EXPECT_TEEC_SUCCESS(c, TEEC_ERROR_OUT_OF_MEMORY);
+		goto out;
+	}
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_MEMREF_TEMP_OUTPUT,
+					 TEEC_MEMREF_TEMP_OUTPUT);
+
+	Do_ADBG_BeginSubCase(c,
+			     "Input/Output MEMREF Buffer NULL - Size 0 bytes");
+
+	op.params[0].tmpref.buffer = empty_buf;
+	op.params[0].tmpref.size = 0;
+
+	op.params[1].tmpref.buffer = NULL;
+	op.params[1].tmpref.size = 0;
+
+	op.params[2].tmpref.buffer = empty_buf;
+	op.params[2].tmpref.size = 0;
+
+	op.params[3].tmpref.buffer = NULL;
+	op.params[3].tmpref.size = 0;
+
+	ADBG_EXPECT(c, TEE_SUCCESS,
+		    TEEC_InvokeCommand(&session, TA_OS_TEST_CMD_PARAMS_TMPMEM,
+				       &op, &ret_orig));
+
+	Do_ADBG_EndSubCase(c, "Input/Output MEMREF Buffer NULL - Size 0 bytes");
+
+	Do_ADBG_BeginSubCase(c, "Input MEMREF Buffer NULL - Size non 0 bytes");
```

Comment:
Sorry, I don"t catch this last test.
The test sets param#0 as a valid buffer (address and size) and invokes test TA returns an error since it expects a valid address and a null size. What does this really test?  I think this test could be removed, only the one above is useful.

Reply:
The test is to verify the buffer is 0 bytes but param size say that buffer is not 0 bytes. Hence a BAD_PARAMETERS error must be returned

Reply:
I don't agree.

This test does not verify that the buffer provided `empty_buf` here relates to 0 bytes or not.
It just check that value set in `tmpref.size` is 0 or not, whatever effective allocated size is used when `empty_buffer` is allocated at line 1895 above.

Note that the effective size of `empty_buffer` is not known by TEE driver or TEE Core (or the TA).
For example, if you allocate 10 byte, and set `.size` to 11 bytes, it is likely that these are already mapped in CA, hence TEE driver/Core will successfully handle the 11 bytes buffer, but that the one that will write to the last byte will corrupt something in CA memory.

Reply:
ok